src/main/webapp/signup.js

```diff
@@ -161,8 +143,30 @@ function createSignupBody() {
   return {
     gitHubAuthToken: gitHubAuthorizer.getToken(),
     interestTags,
-    emailAddress,
   };
 }
 
+/**
+ * Posts a user to the servlet
+ * @param {URLSearchParams} userParams parameters with the users information
+ * @return {?Promise} Returns a prepared post user fetch request.
+ */
+function postUser(userParams) {
+  const fetchRequest = fetch(makeRelativeUrlAbsolute('/user'), {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: userParams,
+  });
+
+  const errorFormattedFetchRequest = standardizeFetchErrors(
+      fetchRequest,
+      'Failed to communicate with the server. Please try again later.',
+      'An error occcured while creating your account.' +
+    ' Please try again later.');
+
+  return errorFormattedFetchRequest;
```

Comment:
Don't store the result of a function call in a variable if it will just be returned in the next line. Just return the result of the function without storing it in a variable first. This is fewer lines of code.
```suggestion
   return  errorFormattedFetchRequest = standardizeFetchErrors(
      fetchRequest,
      'Failed to communicate with the server. Please try again later.',
      'An error occcured while creating your account.' +
    ' Please try again later.');
```